File: padroesEstruturais/padrao-de-projeto-adapter/php/tests/unit/PartidaTest.php
<?php

namespace Adapter\tests\unit;

use PHPUnit\Framework\TestCase;
use Adapter\Php\Adapter\Partida;

class PartidaTest extends TestCase
{
    public function testDeveRetornarDanoPorcentagem(): void
    {
        $partida = new Partida();
        $partida->setDano(1000);
        $this->assertEquals("100%", $partida->getDanoPorcentagem());
    }
}

File: padroesEstruturais/padrao-de-projeto-adapter/java/src/test/java/PartidaTest.java
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class PartidaTest {

    @Test
    public void deveRetornarDanoPorcentagem() {
        Partida partida = new Partida();
        partida.setDano(1000);
        assertEquals("100%", partida.getDanoPorcentagem());
    }

    @Test
    public void deveRetornarDanoNumerico() {
        Partida partida = new Partida();
        partida.setDano(500f);
        assertEquals(500f, partida.getDanoNumerico(), 000.1);
    }
}
